src/wh_client.c

```diff
@@ -308,4 +311,206 @@ int wh_Client_CustomCbCheckRegistered(whClientContext* c, uint16_t id, int* resp
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == NULL)
+        return WH_ERROR_BADARGS;
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+        else
+            *keyId = packet->keyCacheRes.id;
+    }
+    return ret;
+}
+
+int wh_Client_KeyEvictRequest(whClientContext* c, uint16_t keyId)
+{
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == WOLFHSM_ID_ERASED)
+        return WH_ERROR_BADARGS;
+    /* set the keyId */
+    packet->keyEvictReq.id = keyId;
+    /* write request */
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_KEY, WH_KEY_EVICT,
+            WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->keyEvictReq),
+            (uint8_t*)packet);
+}
+
+int wh_Client_KeyEvictResponse(whClientContext* c)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    whPacket packet[1] = {0};
+    if (c == NULL)
+        return WH_ERROR_BADARGS;
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+    }
+    return ret;
+}
+
+int wh_Client_KeyExportRequest(whClientContext* c, uint16_t keyId)
+{
+    whPacket packet[1] = {0};
+    if (c == NULL || keyId == WOLFHSM_ID_ERASED)
+        return WH_ERROR_BADARGS;
+    /* set keyId */
+    packet->keyExportReq.id = keyId;
+    /* write request */
+    return wh_Client_SendRequest(c, WH_MESSAGE_GROUP_KEY, WH_KEY_EXPORT,
+            WOLFHSM_PACKET_STUB_SIZE + sizeof(packet->keyExportReq),
+            (uint8_t*)packet);
+}
+
+int wh_Client_KeyExportResponse(whClientContext* c, uint8_t* label,
+    uint32_t labelSz, uint8_t* out, uint32_t* outSz)
+{
+    uint16_t group;
+    uint16_t action;
+    uint16_t size;
+    int ret;
+    uint8_t rawPacket[WH_COMM_MTU] = {0};
+    whPacket* packet = (whPacket*)rawPacket;
+    uint8_t* packOut = (uint8_t*)(&packet->keyExportRes + 1);
+    if (c == NULL || labelSz > WOLFHSM_NVM_LABEL_LEN || outSz == NULL)
+        return WH_ERROR_BADARGS;
+    ret = wh_Client_RecvResponse(c, &group, &action, &size, rawPacket);
+    if (ret == 0) {
+        if (packet->rc != 0)
+            ret = packet->rc;
+        else  {
+            if (out == NULL) {
+                *outSz = packet->keyExportRes.len;
+            }
+            else if (*outSz < packet->keyExportRes.len) {
+                ret = WH_ERROR_ABORTED;
+            }
+            else {
+                XMEMCPY(out, packOut, packet->keyExportRes.len);
+                *outSz = packet->keyExportRes.len;
+            }
+            if (label != NULL)
+                XMEMCPY(label, packet->keyExportRes.label, labelSz);
+        }
+    }
+    return ret;
+}
+
+int wh_Client_KeyCommitRequest(whClientContext* c, whNvmId keyId)
```

Comment:
I'm confused on why this is an issue, should user facing APIs not use a wolfHSM type?

Reply:
They should, but we are talking about an NvmId rather than a KeyId.  Both happen to be uint16_t's but the spelling should be distinct and the choice of type should be explicit.  I updated the backlog item to create a whKeyId type.